src/client/main.tsx

```diff
@@ -24,22 +34,67 @@ export class Main extends React.Component<{}, ClientState> {
       console.log('connection!');
       this.startGame(initialData.gameState, initialData.playerId);
     });
-    this.setState({ stage: Stages.LOADING });
+    this.socket.on('new player', (numPlayers: number) => {
+      this.resetTimer();
+      this.updateCountdown(GAME_START_TIME);
+      this.setState({
+        stage: this.state.stage,
+        numPlayersInLobby: numPlayers,
+        countdownTime: GAME_START_TIME
+      });
+    })
+    this.setState({
+      stage: Stages.LOADING,
+      numPlayersInLobby: this.state.numPlayersInLobby,
+      countdownTime: this.state.countdownTime
+    });
   }
 
   startGame(initialState: GameState, playerId: string) {
       this.gameState = initialState;
       this.activePlayer = playerId;
       this.controller = new ClientController(this.gameState, this.socket);
-      this.setState({stage: Stages.RUNNING});
+      this.resetTimer();
+      this.setState({
+        stage: Stages.RUNNING,
+        numPlayersInLobby: this.state.numPlayersInLobby,
+        countdownTime: this.state.countdownTime
+      });
+  }
+
+  updateCountdown(time: number) {
+    this.setState({
+      stage: this.state.stage,
+      numPlayersInLobby: this.state.numPlayersInLobby,
+      countdownTime: time
+    });
+    if (time - 1 >= 0) {
+      this.countdownTimer = setTimeout(this.updateCountdown.bind(this), 1000, time - 1);
+    }
+  }
+
+  resetTimer() {
```

Comment:
Note: Once this lobby stuff is in a separate react component, this probably won't be necessary